From 2bfe68e84e19165ac8b072cf49c187ae7cf81802 Mon Sep 17 00:00:00 2001
From: Arik Nemtsov <arik@wizery.com>
Date: Thu, 13 Sep 2012 14:23:45 +0300
Subject: [PATCH 38/40] nl80211: sync nl80211.h with latest wpa_s copy

Also move channel switch related command to the TI private commands
range.

Signed-off-by: Arik Nemtsov <arik@wizery.com>
---
 drivers/net/wireless/compat-wireless_R5.SP2.03/include/linux/nl80211.h |   24 +++++++++++++-----------
 1 files changed, 13 insertions(+), 11 deletions(-)

diff --git a/drivers/net/wireless/compat-wireless_R5.SP2.03/include/linux/nl80211.h b/drivers/net/wireless/compat-wireless_R5.SP2.03/include/linux/nl80211.h
index fa07bc0..fbe3733 100644
--- a/drivers/net/wireless/compat-wireless_R5.SP2.03/include/linux/nl80211.h
+++ b/drivers/net/wireless/compat-wireless_R5.SP2.03/include/linux/nl80211.h
@@ -565,14 +565,6 @@
  *	%NL80211_ATTR_IFINDEX is now on %NL80211_ATTR_WIPHY_FREQ with
  *	%NL80211_ATTR_WIPHY_CHANNEL_TYPE.
  *
-* @NL80211_CMD_AP_CH_SWITCH: Perform a channel switch in the driver (for
- *	AP/GO).
- *	%NL80211_ATTR_WIPHY_FREQ: new channel frequency.
- *	%NL80211_ATTR_CH_SWITCH_BLOCK_TX: block tx on the current channel.
- *	%NL80211_ATTR_CH_SWITCH_POST_BLOCK_TX: block tx on the target channel.
- *	%NL80211_FREQ_ATTR_CH_SWITCH_COUNT: number of TBTT's until the channel
- *	switch event.
- *
  * @NL80211_CMD_SCAN_CANCEL: Stop currently running scan (both sw and hw).
  *	This operation will eventually invoke %NL80211_CMD_SCAN_ABORTED
  *	event, partial scan results will be available. Returns -ENOENT
@@ -591,6 +583,14 @@
  *      disabled (marked by the presence of @NL80211_ATTR_ROAMING_DISABLED flag)
  *      userspace should disable background scans and roaming attempts.
  *
+ * @NL80211_CMD_AP_CH_SWITCH: Perform a channel switch in the driver (for
+ *	AP/GO).
+ *	%NL80211_ATTR_WIPHY_FREQ: new channel frequency.
+ *	%NL80211_ATTR_CH_SWITCH_BLOCK_TX: block tx on the current channel.
+ *	%NL80211_ATTR_CH_SWITCH_POST_BLOCK_TX: block tx on the target channel.
+ *	%NL80211_FREQ_ATTR_CH_SWITCH_COUNT: number of TBTT's until the channel
+ *	switch event.
+ *
  * @NL80211_CMD_REQ_CH_SW: Request a channel switch from a GO/AP.
  *
  * @NL80211_CMD_MAX: highest used command number
@@ -736,8 +736,6 @@ enum nl80211_commands {
 
 	NL80211_CMD_CH_SWITCH_NOTIFY,
 
-	NL80211_CMD_AP_CH_SWITCH,
-
 	/* leave some room for adding nl80211 commands for old kernels */
 	NL80211_CMD_SCAN_CANCEL = NL80211_CMD_CH_SWITCH_NOTIFY + 10,
 
@@ -745,6 +743,10 @@ enum nl80211_commands {
 
 	NL80211_CMD_ROAMING_SUPPORT,
 
+	NL80211_CMD_SET_PRIORITY,
+	NL80211_CMD_CANCEL_PRIORITY,
+
+	NL80211_CMD_AP_CH_SWITCH,
 	NL80211_CMD_REQ_CH_SW,
 
 	/* add new commands above here */
@@ -3128,10 +3130,10 @@ enum nl80211_feature_flags {
 	NL80211_FEATURE_HT_IBSS		= 1 << 1,
 	NL80211_FEATURE_INACTIVITY_TIMER = 1 << 2,
 	NL80211_FEATURE_CELL_BASE_REG_HINTS = 1 << 3,
-	NL80211_FEATURE_AP_CH_SWITCH	= 1 << 4,
 
 	/* leave room for new feature flags */
 	NL80211_FEATURE_SCHED_SCAN_INTERVALS  = 1 << 20,
+	NL80211_FEATURE_AP_CH_SWITCH	= 1 << 21,
 };
 
 /**
-- 
1.7.0.4

